js/graviex.js

```diff
@@ -0,0 +1,1012 @@
+            const firstTicker = this.safeValue (response, 0, {});
+            return this.parseTicker (firstTicker, market);
+        }
+        return this.parseTicker (response, market);
+    }
+
+    async fetchTickers (symbols = undefined, params = {}) {
+        await this.loadMarkets ();
+        const response = await this.publicGetTickers (params);
+        // {
+        //   "giobtc": {
+        //     "name": "GIO/BTC",
+        //     "base_unit": "gio",
+        //     "base_fixed": 4,
+        //     "base_fee": 0.002,
+        //     "quote_unit": "btc",
+        //     "quote_fixed": 8,
+        //     "quote_fee": 0.002,
+        //     "api": true,
+        //     "base_lot": null,
+        //     "quote_lot": null,
+        //     "base_min": "0.00000010",
+        //     "quote_min": "0.00000010",
+        //     "blocks": 260722,
+        //     "block_time": "2022-01-31 14:10:08",
+        //     "wstatus": "on",
+        //     "low": "0.0000005",
+        //     "high": "0.00000058",
+        //     "last": "0.00000058",
+        //     "open": "0.000000510",
+        //     "volume": "70673.0412",
+        //     "volume2": "0.03837782973",
+        //     "sell": "0.00000057",
+        //     "buy": "0.00000055",
+        //     "at": 1643628325,
+        //   },
+        // },
+        // const data = response;
+        // let timestamp = data['at'];
+        // let tickers = data['ticker'];
+        const ids = Object.keys (response);
+        const result = {};
+        for (let i = 0; i < ids.length; i++) {
+            const id = ids[i];
+            const market = this.safeMarket (id);
+            const symbol = market['symbol'];
+            result[symbol] = this.parseTicker (response[id], market);
+        }
+        return result;
+    }
+
+    parseTrade (trade, market = undefined) {
+        const price = this.safeString2 (trade, 'p', 'price');
+        const amount = this.safeString2 (trade, 'volume', 'amount');
+        const funds = this.safeString (trade, 'funds');
+        const marketId = this.safeString (trade, 'symbol');
+        const symbol = this.safeSymbol (marketId, market);
+        const datetime = this.safeValue (trade, 'created_at');
+        let timestamp = this.parse_date (datetime);
+        let id = this.safeString2 (trade, 't', 'a');
+        id = this.safeString2 (trade, 'id', 'tid', id);
+        const side = this.safeValue (trade, 'side');
+        const fee = undefined;
+        const cost = undefined;
+        let type = undefined;
+        let takerOrMaker = undefined;
+        const orderId = this.safeString (trade, 'id');
+        if ('side' in trade) {
+            if (trade['side'] === 'buy') {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        if ('date' in trade) {
+            timestamp = this.safeInteger2 (trade, 'T', 'date');
+        }
+        if ('type' in trade) {
```

Comment:
First  `if ('type' in trade) ` is evaluated twice unnecessarily, second can you please point me to the docs where this takerOrMaker is stated? I'm not sure if this is the trade position only or the takerOrMaker side as is right now 🙂

Reply:
Ok. Fixed in the latest commit